src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -423,31 +423,19 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
             cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgs))
             return report
 
-        reqArgsNothandled = []
-        for reqArg in reqArgs:
-            if reqArg == 'RequestPriority':
-                validate_request_priority(reqArgs)
-                # must update three places: GQ elements, workload_cache and workload spec
-                self.gq_service.updatePriority(workload.name(), reqArgs['RequestPriority'])
-                workload.setPriority(reqArgs['RequestPriority'])
-                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgs['RequestPriority']))
-            elif reqArg == "SiteWhitelist":
-                workload.setSiteWhitelist(reqArgs["SiteWhitelist"])
-                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteWhitelist']))
-            elif reqArg == "SiteBlacklist":
-                workload.setSiteBlacklist(reqArgs["SiteBlacklist"])
-                cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgs['SiteBlacklist']))
-            else:
-                reqArgsNothandled.append(reqArg)
-                cherrypy.log("Unhandled argument for no-status update: %s" % reqArg)
+        # Update all workload parameters based on the full reqArgs dictionary
+        workload.updateWorkloadArgs(reqArgs)
 
-        if reqArgsNothandled:
-            msg = "There were unhandled arguments left for no-status update: %s" % reqArgsNothandled
-            raise InvalidSpecParameterValue(msg)
+        # Commit the changes of the current workload object to the database:
+        workload.saveCouchUrl(workload.specUrl())
 
         # Commit the changes of the current workload object to the database:
         workload.saveCouchUrl(workload.specUrl())
 
+        # Commit all Global WorkQueue changes per workflow in a single go:
+        self.gq_service.updateElementsByWorkflow(workload.name(), reqArgs, status=['Available'])
```

Comment:
> With this call, you will actually change the behavior of priority to get only applied to Available elements.

Well, does it even matter? As we  discussed with you few  days ago - do we actually know if we continue to update any Worqueue Element's parameters at the agent once it is acquired from the GlobalWorkQueue ? If we do, then it matters also for Sitelists as well. So IIRC, during our discussion we decided to stay safe and update only available WQE and not to risk getting into some race conditions related to input data location  etc. If we do not care I can remove the requirement for `status=['Availiable']`. But if we care to propagate the `Priority` information down to the agent for every WQE status - `Aqcuired et. all.`, but we want to update site information only at Global WQ, then we simply  cannot have a mechanism to do it in a single push. The two calls should be separated. So we cannot have them both:
* a single DB action for updating WQE
* different WQE status requirement based on request parameters
(an escape of this is to hardcode this in the Workqueue module logic of `updateelementsByWorkflow`, but you know my opinion on hardcoding requirements in the code.

> In addition, do you ensure now that reqArgs will always be only the supported parameters?

As long as we follow the same logic as we did while validating the allowed parameters in the first set of actions in these calls (meaning to call `validate_request_update_args` before calling the `put` method and hence `_handleNoStatusUpdate`) then we are keeping the same logic as before - hence following the map of allowed actions  from https://github.com/dmwm/WMCore/blob/7ddd036d68d43f3018a6215ff63381fe098fd1fe/src/python/WMCore/ReqMgr/DataStructs/RequestStatus.py#L118-L147. And for this logic we needed mostly those 3 setters methods which I've already put in the `reqArgs` to `setterMethod` map at `updateWorkloadArgs` :
```
self.settersMap['RequestPriority'] = setterTuple('RequestPriority', self.setPriority, inspect.signature(self.setPriority))
self.settersMap['SiteBlacklist'] = setterTuple('SiteBlacklist', self.setSiteBlacklist, inspect.signature(self.setSiteBlacklist))
self.settersMap['SiteWhitelist'] = setterTuple('SiteWhitelist', self.setSiteWhitelist, inspect.signature(self.setSiteWhitelist))
```   

If we  want to be extra detailed and fully exhaustive on any possible `reqarg` per `status` setters, we may also include into this map the eventual methods for all allowed actions per `assignment-approved` status.:
```
    "assignment-approved": ["RequestPriority", "Team", "SiteWhitelist", "SiteBlacklist",
                            "AcquisitionEra", "ProcessingString", "ProcessingVersion",
                            "Dashboard", "MergedLFNBase", "TrustSitelists",
                            "UnmergedLFNBase", "MinMergeSize", "MaxMergeSize",
                            "MaxMergeEvents", "BlockCloseMaxWaitTime",
                            "BlockCloseMaxFiles", "BlockCloseMaxEvents", "BlockCloseMaxSize",
                            "SoftTimeout", "GracePeriod",
                            "TrustPUSitelists", "CustodialSites",
                            "NonCustodialSites", "Override",
                            "SubscriptionPriority"],
 ```   

But for some of them the logic may need to change because some of those setters are parametrized by more than a single argument. While at the same time we already know that for `assignment-approved` we would never call `_handleNoStatusUpdate` ... So that's why, to me it seems safe to proceed only with those 3 setters mapped. Of course,  if you ask me - I am completely up for moving the whole logic to be implemented here in a more generic way .... for all status updates, then get rid of a big chunk of code covering custom cases ... and only make the proper calls to this generic method here from upstream modules (e.g. Request in the current case) But I do not think we will have the time during this line of development  here to do this.

Reply:
We might be able to distribute this update, but before answering how we can proceed, can you please check the JobUpdater code to check:
1) in which databases are the WQEs updated?
2) does it use any status filter? If so, which statuses?

Reply:
hi @amaltaro, 
ok, Here it is:
* `JobUpdater` at the agent  uses the very same method `updatePriority` https://github.com/dmwm/WMCore/blob/4e0759d5648dfb66de7d7bc135e2975f88fecfbf/src/python/WMCore/Services/WorkQueue/WorkQueue.py#L240 

from the `WMCore.Services.WorkQueue` to update the workflow priority (just as before), here is the actual call:
https://github.com/dmwm/WMCore/blob/4e0759d5648dfb66de7d7bc135e2975f88fecfbf/src/python/WMComponent/JobUpdater/JobUpdaterPoller.py#L151

* It checks for any diff between the priority in WMBS and couch database before taking actions:
https://github.com/dmwm/WMCore/blob/4e0759d5648dfb66de7d7bc135e2975f88fecfbf/src/python/WMComponent/JobUpdater/JobUpdaterPoller.py#L149
and:
https://github.com/dmwm/WMCore/blob/4e0759d5648dfb66de7d7bc135e2975f88fecfbf/src/python/WMComponent/JobUpdater/JobUpdaterPoller.py#L166

* The couchd database to be used is taken from the agent configuration for the component (actually the regular local workqueue component, not the `JobUpdater` config section): 
https://github.com/dmwm/WMCore/blob/4e0759d5648dfb66de7d7bc135e2975f88fecfbf/src/python/WMComponent/JobUpdater/JobUpdaterPoller.py#L50-L51  

which in this case is (taken from one agent configuration):
```
config.WorkQueueManager.dbname = 'workqueue'
```

So the inbox db is indeed configured at the `WorkqueueManager` component at  the agent, but  as  long as I do not see any actual actions  during this update from the `updatepriority` method, I do not think we even touch this DB while updating the WQE priority at the agents.